backend/pkg/handler/groupHandler.go

```diff
@@ -113,38 +116,94 @@ func (h *GroupHandler) EditGroupHandler(w http.ResponseWriter, r *http.Request)
 // TODO: implement notification to all group members that the group has been deleted, and remove all group members;
 // implement logging of the deletion or add bool field "deleted"
 func (h *GroupHandler) DeleteGroupHandler(w http.ResponseWriter, r *http.Request) {
-    userID, err := h.sessionRepo.GetUserIDFromSessionToken(util.GetSessionToken(r))
-    if err != nil {
-        http.Error(w, "Error confirming authentication: " + err.Error(), http.StatusInternalServerError)
-        return
-    }
-
-    // logic for deleting a group
-    vars := mux.Vars(r)
-    id, err := strconv.Atoi(vars["id"])
-    if err != nil {
-        http.Error(w, "Invalid group ID", http.StatusBadRequest)
-        return
-    }
-    
-    group, err := h.groupRepo.GetGroupByID(id)
-    if err != nil {
-        http.Error(w, "Failed to get group: "+err.Error(), http.StatusInternalServerError)
-        return
-    }
-    if group.CreatorId != userID {
-        http.Error(w, "User not authorized to delete this group", http.StatusUnauthorized)
-        return
-    }
-    err = h.groupRepo.DeleteGroup(id)
-    if err != nil {
-        http.Error(w, "Failed to delete group: "+err.Error(), http.StatusInternalServerError)
-        return
-    }
-    // Successful response
+	userID, err := h.sessionRepo.GetUserIDFromSessionToken(util.GetSessionToken(r))
+	if err != nil {
+		http.Error(w, "Error confirming authentication: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// logic for deleting a group
+	vars := mux.Vars(r)
+	id, err := strconv.Atoi(vars["id"])
+	if err != nil {
+		http.Error(w, "Invalid group ID", http.StatusBadRequest)
+		return
+	}
+
+	group, err := h.groupRepo.GetGroupByID(id)
+	if err != nil {
+		http.Error(w, "Failed to get group: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if group.CreatorId != userID {
+		http.Error(w, "User not authorized to delete this group", http.StatusUnauthorized)
+		return
+	}
+
+	// Notify all group members that the group has been deleted
+	err = notifyGroupDeletion(h.groupMemberRepo, h.notificationRepo, id)
+	if err != nil {
+		http.Error(w, "Failed to notify group members about group deletion: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Remove all group members
```

Comment:
no need to remove the group members since the foreign keys of the groupID in group_members table has ON DELETE CASCADE